src/utils/barn.ts

```diff
@@ -139,3 +150,42 @@ export const oppdaterBarnLabels = (barn: IBarn[], intl: LokalIntlShape) => {
 
   return oppdaterteBarn;
 };
+
+const unikeForelderidenter = (barn: IBarn[]) =>
+  Array.from(new Set(barn.map((b) => b.forelder?.ident?.verdi))).filter(
+    (forelderIdent): forelderIdent is string => forelderIdent !== undefined
```

Comment:
Hva gjør `is string` her?

Reply:
Er en type guard som gjør at vi kan filtrere bort undefined. Uten den skjønner ikke typescript at listen ikke inneholder `undefined`

Reply:
Men den inneholder jo undefined, som filtreres bort? 😂

Reply:
```js
const unikeForelderidenter = (barn: IBarn[]) =>
  Array.from(new Set(barn.map((b) => b.forelder?.ident?.verdi))).filter(
    (forelderIdent): forelderIdent is string => forelderIdent !== undefined
  );

// returtype i følge typescript: string[]

const utenTypeGuard = (barn: IBarn[]) => //
  Array.from(new Set(barn.map((b) => b.forelder?.ident?.verdi))).filter(
    (forelderIdent) => forelderIdent !== undefined
  );

// returtype i følge typescript: (string | undefined)[]
```

Uten type guarden må vi legge på nullsjekker når vi itererer over lista

Reply:
Trukket ut som util-funksjon for å gjøre lettere å forstå i https://github.com/navikt/familie-ef-soknad/pull/1357/commits/c6598076664f1a276823248a23adbb280e2917bd